packages/core/test/unit/query-generator/create-table-query.test.ts

```diff
@@ -0,0 +1,463 @@
+import { createSequelizeInstance, expectsql, sequelize } from '../../support';
+
+const dialect = sequelize.dialect;
+
+// TODO: check the tests with COMMENT after attributeToSQL quotes the comment
+// TODO: double check if all column SQL types are possible results of attributeToSQL after #15533 has been merged
+// TODO: see if some logic in handling columns can be moved to attributeToSQL which could make some tests here redundant
+
+describe('QueryGenerator#createTableQuery', () => {
+  const queryGenerator = sequelize.getQueryInterface().queryGenerator;
+
+  it('produces a query to create a table', () => {
+    expectsql(queryGenerator.createTableQuery('myTable', { myColumn: 'DATE' }), {
+      default: 'CREATE TABLE IF NOT EXISTS [myTable] ([myColumn] DATE);',
+      'mariadb mysql': 'CREATE TABLE IF NOT EXISTS `myTable` (`myColumn` DATE) ENGINE=InnoDB;',
+      mssql: `IF OBJECT_ID(N'[myTable]', 'U') IS NULL CREATE TABLE [myTable] ([myColumn] DATE);`,
+      ibmi: `BEGIN DECLARE CONTINUE HANDLER FOR SQLSTATE VALUE '42710' BEGIN END; CREATE TABLE "myTable" ("myColumn" DATE); END`,
+    });
+  });
+
+  it('produces a query to create a table from a model', () => {
+    const MyModel = sequelize.define('myModel', {});
```

Comment:
Same comment as on the other PR: We should name our models like classes

```suggestion
    const MyModel = sequelize.define('MyModel', {});
```

Reply:
Updated